lib/readwise/document.rb

```diff
@@ -0,0 +1,140 @@
+require 'time'
+
+module Readwise
+  Document = Struct.new(
+    'ReadwiseDocument',
+    :author,
+    :category,  # One of: article, email, rss, highlight, note, pdf, epub, tweet or video.
+                # Default is guessed based on the URL, usually article.
+    :created_at,
+    :html,
+    :id,
+    :image_url,
+    :location,  # One of: new, later, archive or feed. Default is new.
+    :notes,
+    :published_date,
+    :reading_progress,
+    :site_name,
+    :source,
+    :source_url,
+    :summary,
+    :tags,
+    :title,
+    :updated_at,
+    :url,
+    :word_count,
+    :parent_id, # both highlights and notes made in Reader are also considered Documents.
+                # Highlights and notes will have `parent_id` set, which is the Document id
+                # of the article/book/etc and highlight that they belong to, respectively.
+    keyword_init: true
```

Comment:
btw if you can choose to only support Ruby 3.2 or newer then it defaults to `keyword_init`.